src/masquer/utils/sec_ch.py

```diff
@@ -0,0 +1,56 @@
+from .assets import (SEC_CH_UA, HEADER_DATA)
+import re
+def append_sec_ch(user_agent: str, headers: dict) -> str:
+    if(user_agent.find('Chrome') == -1):
+        return headers
+    platform = get_platform(user_agent)
+    headers['sec-ch-ua-platform'] = platform
+    headers['sec-ch-ua-mobile'] = "?1" if is_mobile(user_agent) else "?0"
+    headers['sec-ch-ua'] = find_sec_ch(get_chrome_version(user_agent)).replace('%c', get_browser(user_agent))
```

Comment:
I may have read this incorrectly, but it seems that if "Not Found" is returned by `find_sec_ch` because the version doesn't exist in the `SEC_CH_UA` dict, that will be the final value for `sec-ch-ua` in the header?

If that's correct, is it valid to send this? 

Whether it is or isn't valid, perhaps the approach should be to first check whether the Chromium version is in the dict before adding any of these attributes &mdash; if not, then revert to the default headers as per lines 4 & 5?

It takes me a bit of time to keep the values in my head with method chaining, please consider unpacking that last line into separate variables:

```py
browser = get_browser(useragent)
version = get_chrome_version(useragent)
sec_ch_ua = find_sec_ch(version)
sec_ch_ua = sec_ch_ua.replace('%c', browser)
headers['sec-ch-ua'] = sec_ch_ua
```

Reply:
It isn't correct, but neither this nor completely removing it will be valid, so I thought this might be a viable solution if the anti-bot solution is just checking for the header existence instead of checking if it has the correct value.

as far as I know, Amazon Cloudfront(as an example) has some kind of self-training, where they start by allowing the request if the header just exists, and starts getting more strict as the chrome version gets older/user IP makes multiple requests using same set of headers.

Sure thing, Sorry for that, I'm not good at python.

Reply:
Thanks - the code is fine, it's just my limited reading ability. 😄

OK noted on the header checks, that's really interesting. I'll read more into this. If the main `sec-ch-ua` are captured in the options then hopefully it should rarely pose a problem anway.